orchestrator/test/test_application_class.py

```diff
@@ -114,14 +116,36 @@ def test_monkey(self, device, support_app):  # noqa
         finally:
             app.uninstall()
 
-    def test_clear_data(self, device, support_app):  # noqa
+    def test_clear_data(self, device: Device, support_app: str):  # noqa
         uninstall_apk(support_app, device)
         app = Application.from_apk(support_app, device)
         try:
             app.clear_data()  # should not raise exception
         finally:
             app.uninstall()
 
-    def test_version_invalid_package(self, device):
+    def test_version_invalid_package(self, device: Device):
         with pytest.raises(Exception):
             Application.from_apk("no.such.package", device)
+
+    def test_app_uninstall_logs_error(self, device: Device):
+        with patch("androidtestorchestrator.application.log") as mock_logger:
+            app = Application(package_name="com.android.providers.calendar", device=device)
+            app.uninstall()
+            assert mock_logger.error.called
+
+    def test_clean_kill(self, device: Device, support_app: str):
+        app = Application.from_apk(support_app, device)
```

Comment:
Possibly could move the "install app; try: finally uninstall" bit into a fixture.

Reply:
I can address that in a future commit. Want to get this one in for now.

Reply:
Yup, agreed.